### PR TITLE
Add error load log url for routine load job

### PR DESCRIPTION
**Pull request by morningman**

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 1]
```diff
@@ -574,32 +585,38 @@ public void beforeCommitted(TransactionState txnState) throws TransactionExcepti
...
-            // The commit txn by thread2 will be failed after waiting for timeout seconds.
-            // Maybe thread1 hang on somewhere
-            LOG.warn(new LogBuilder(LogKey.ROUINTE_LOAD_TASK, txnState.getLabel())
-                             .add("job_id", id)
-                             .add("txn_status", transactionStatus)
-                             .add("error_msg", "txn could not be transformed "
-                                     + "while waiting for timeout of routine load job"));
-            throw new TransactionException("txn " + txnState.getTransactionId() + "could not be " + transactionStatus
-                                                   + "while waiting for timeout of routine load job.", e);
+        } catch (Throwable e) {
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: Why you catch the Throwable in here ?

File: fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 2]
```diff
@@ -925,31 +946,37 @@ public void setOrigStmt(String origStmt) {
...
-        switch (state) {
-            case PAUSED:
-                row.add(pausedReason);
-                break;
-            case CANCELLED:
-                row.add(cancelReason);
-                break;
-            default:
-                row.add("");
+        readLock();
```
Comment (EmmyMiao87) [inline fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: The strong consistency need be protected in here?

File: fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: Changed it to TransactionException

File: fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java (hunk 2 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: Yes, the errorLogUrls may be modified anytime

**Pull request merged**